api/client/client.go

```diff
@@ -38,23 +42,24 @@ type Remote interface {
 	Hosts() []string
 }
 
-// NewServer sets up a new server target. The address should be the
-// DNS name (or "name:port") of the remote CFSSL instance. If no port
+// NewServer sets up a new server target. The address should be of
+// The format [protocol:]name[:port] of the remote CFSSL instance.
+// If no protocol is given http is default. If no port
 // is specified, the CFSSL default port (8888) is used. If the name is
 // a comma-separated list of hosts, an ordered group will be returned.
-func NewServer(addr string) Remote {
-	addrs := strings.Split(addr, ",")
+func NewServer(addr string, remoteCAs *x509.CertPool, cert *tls.Certificate) Remote {
```

Comment:
I might just add `NewServeTLSr` and `NewGroupTLS` and keep the old API the same signature. `NewServer` is then just:

```Go
func NewServer(addr string) Remote {
	return NewServerTLS(addr, nil)
}
```